architecture/kubernetes_infrastructure.adoc

```diff
@@ -9,30 +9,43 @@
 
 toc::[]
 
-== Master
+Kubernetes is a system for managing containerized applications across a set of containers or hosts, providing deployment, maintenence, and application-scaling mechanisms. Docker is used to package, instantiate, and run containerized applications. 
 
-The host(s) containing the master components (API Server, Controller Manger Server, etcd, etc)
+A Kubernetes cluster involves a master and a set of nodes. 
 
-=== API Server
+== Master
 
-Serves the main API
+The Kubernetes master is the host or hosts that contain the master components (API Server, Controller Manger Server, etcd, etc)
 
-=== Controller Manager Server
+=== API Server
 
-Watches etcd for changes to ReplicationController objects and then uses the API to enforce the desired state.
+The Kubernetes API server serves the main API, and validates and configures the data for pods, services, and replication controllers. The Kuberbetes API server also assigns pods to nodes and synchronizes pod information with service configuration. 
 
 === etcd
 
 All persistent master state is stored in etcd. Other components watch etcd for changes and eventually bring themselves into the desired state described in etcd.
 
+=== Controller Manager Server
+
+The controller manager server watches etcd for changes to replication controller objects and then uses the API to enforce the desired state.
+
 == Node
 
-Each node has the services necessary to run Docker containers and be managed from the master systems
+A Kubernetes node is managed from the master, and has the services necessary in order to run Docker containers. Each node runs Docker, which downloads and runs containers.
+
+A node is not created by Kubernetes, it is instead created by your cloud provider, or from your physical or virtual machines. Once Kubernetes is involved, it will validate the node with health checks. A node will be ignored until these health checks are passed. Note that Kubernetes will keep checking nodes until they are valid. 
```

Comment:
@bfallonf "Once Kubernetes is involved, it will validate the node with health checks. A node will be ignored until these health checks are passed. Note that Kubernetes will keep checking nodes until they are valid. "

Kubernetes validates the node with health checks and ignores it until these health checks are passed. Note that Kubernetes continues to check nodes until they are valid.